service-loadbalancer/service_loadbalancer.go

```diff
@@ -387,9 +420,17 @@ func (lbc *loadBalancerController) getServices() (httpSvc []service, tcpSvc []se
 					sName, servicePort)
 				continue
 			}
+
+			rules := getLBRules(lbc, &s, servicePort.Port)
```

Comment:
please pass in servicePort instead of servicePort.Port, since we also have named ports.